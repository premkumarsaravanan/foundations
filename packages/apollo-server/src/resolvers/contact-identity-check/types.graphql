
# Types
type ContactIdentityDocument {
  typeId: String
  expiry: String
  details: String
  fileUrl: String
  _links: JSON
  _embedded: JSON
}

type ContactIdentityCheck {
  id: String
  contactId: String
  created: String
  modified: String
  checkDate: String
  status: String
  negotiatorId: String
  documents: [ContactIdentityDocument!]
  metadata: JSON
  _links: JSON
  _embedded: JSON
}

type ContactIdentityChecks {
  contactIdentityChecks: [ContactIdentityCheck!]
  pageNumber: Int
  pageSize: Int
  pageCount: Int
  totalCount: Int
  _links: JSON
}

# Input Types
input ContactIdentityDocumentInput {
  typeId: String
  expiry: String
  details: String
  fileUrl: String
}

input ContactIdentityCheckInput {
  checkDate: String
  status: String
  negotiatorId: String
  documents: [ContactIdentityDocumentInput!]
  metadata: JSON
}
