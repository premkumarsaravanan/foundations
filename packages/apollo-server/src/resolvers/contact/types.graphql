# Types
type ContactAddress {
  type: String
  buildingName: String
  buildingNumber: String
  line1: String
  line2: String
  line3: String
  line4: String
  postcode: String
  countryId: String
}

type ContactCommunication {
  label: String
  detail: String
}

type Contact {
  id: String
  created: String
  modified: String
  title: String
  forename: String
  surname: String
  dateOfBirth: String
  active: Boolean
  marketingConsent: String
  identityCheck: String
  communications: [ContactCommunication!]
  addresses: [ContactAddress!]
  officeIds: [String!]
  negotiatorIds: [String!]
  metadata: JSON
  _links: JSON
  _embedded: JSON
}

type Contacts {
  _embedded: [Contact!]
  pageNumber: Int
  pageSize: Int
  pageCount: Int
  totalCount: Int
  _links: JSON
}

# Input Types
input ContactCommunicationInput {
  label: String
  detail: String
}

input ContactAddressInput {
  type: String
  buildingName: String
  buildingNumber: String
  line1: String
  line2: String
  line3: String
  line4: String
  postcode: String
  countryId: String
}
