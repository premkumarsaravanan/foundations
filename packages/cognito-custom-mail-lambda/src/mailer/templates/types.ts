export interface ForgottenPasswordTemplateParams {
  verificationCode: string
  userName: string
  url: string
}

export interface ConfirmPasswordTemplateParams {
  userName: string
  url: string
}
