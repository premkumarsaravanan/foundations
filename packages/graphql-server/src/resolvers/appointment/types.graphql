# Types
type AppointmentContact {

  """
  The unique identifier of the contact
  """
  id: String

  """
  The name of the contact
  """
  name: String

  """
  The home phone number of the contact
  """
  homePhone: String

  """
  The work phone number of the contact
  """
  workPhone: String

  """
  The mobile phone number of the contact
  """
  mobilePhone: String

  """
  The email address of the contact
  """
  email: String
}

type AppointmentAttendee {
  """
  The unique identifier of the attendee
  """
  id: String

  """
  The type of attendee
  """
  type: String

  """
  A collection of contacts relating to the attendee
  """
  contacts: [AppointmentContact]
}

type AppointmentFollowUp {
  """
  The date when the appointment should be followed up
  """
  due: String

  """
  The unique identifier of a pre-defined follow up response type
  """
  responseId: String

  """
  Free text internal follow up notes to be stored against the appointment
  """
  notes: String
}

type Appointment {
  """
  The unique identifier of the appointment
  """
  id: String

  """
  The date and time when the appointment was created
  """
  created: String

  """
  The date and time when the appointment was last modified
  """
  modified: String

  """
  The date and time when the appointment will start
  """
  start: String

  """
  The date and time when the appointment will end
  """
  end: String

  """
  The unique identifier of the appointment type
  """
  typeId: String

  """
  A free text description about the appointment
  """
  description: String

  """
  A flag denoting whether or not the appointment recurs
  """
  recurring: Boolean

  """
  A flag denoting whether or not the appointment has been cancelled
  """
  cancelled: Boolean

  """
  Follow up information relating to the appointment
  """
  followUp: AppointmentFollowUp

  """
  The unique identifier of the property related to the appointment
  """
  propertyId: String

  """
  The unique identifier of the negotiator that organised the appointment
  """
  organiserId: String

  """
  A collection of unique identifiers of negotiators attached to the appointment
  """
  negotiatorIds: [String!]

  """
  A collection of unique identifiers of offices attached to the appointment
  """
  officeIds: [String!]

  """
  An appointment attendee
  """
  attendee: AppointmentAttendee

  """
  A flag denoting whether or not the appointment will be accompanied by one or more negotiators
  """
  accompanied: Boolean

  """
  A flag denoting whether or not the main negotiator has confirmed their attendance
  """
  negotiatorConfirmed: Boolean

  """
  A flag denoting whether or not the attendee has confirmed their attendance
  """
  attendeeConfirmed: Boolean

  """
  A flag denoting whether or not the property and/or property's vendor has confirmed their attendance
  """
  propertyConfirmed: Boolean

  """
  App specific metadata that has been set against the appointment
  """
  metadata: JSON

  """
  The ETag for the current version of the appointment. Used for managing update concurrency
  """
  _eTag: String
  _links: JSON
  _embedded: JSON
}

type GetAppointmentsResult {
  pageNumber: Int
  pageSize: Int
  pageCount: Int
  totalCount: Int
  _links: JSON,
  _embedded: [Appointment]
}


input UpdateAppointmentInputAttendee {
    """The unique identifier of the attendee"""
  id: String

  """The type of attendee (applicant/contact/landlord/tenant)"""
  type: String

  """
  A flag denoting whether or not the attendee has confirmed their attendance
  """
  confirmed: Boolean
}

# Input Types
input CreateAppointmentInputAttendee {
    """The unique identifier of the attendee"""
  id: String

  """The type of attendee (applicant/contact/landlord/tenant)"""
  type: String
}

input CreateAppointmentInputRecurrence {
    """The numeric value denoting how often the appointment will recur"""
  interval: Int

  """The type of unit that the `interval` applies to"""
  type: String

  """The date and time when the recurrence will stop"""
  until: String
}

input UpdateAppointmentInputRecurrence {
  """The type of unit that the `interval` applies to"""
  type: String

  """The numeric value denoting how often the appointment will recur"""
  interval: Int

  """The date and time when the recurrence will stop"""
  until: String
}

input UpdateAppointmentInputFollowUp {
  """The unique identifier of a pre-defined follow up response type"""
  responseId: String

  """The internal follow up notes to be stored against the appointment"""
  notes: String
}
