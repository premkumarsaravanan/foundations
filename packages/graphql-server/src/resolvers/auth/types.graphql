enum LoginMode {
  DESKTOP
  WEB
}

enum LoginType {
  DEVELOPER
  CLIENT
  ADMIN
}

type Token {
  accessToken: String!,
  refreshToken: String!,
}