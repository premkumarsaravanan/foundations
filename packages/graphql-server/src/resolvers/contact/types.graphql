# Types
type ContactAddress {
  type: String
  buildingName: String
  buildingNumber: String
  line1: String
  line2: String
  line3: String
  line4: String
  postcode: String
  countryId: String
}

type ContactCommunication {
  label: String
  detail: String
}

type ContactSource {
  id: String
  type: String
}

type Contact {
  id: String
  created: String
  modified: String
  title: String
  forename: String
  surname: String
  dateOfBirth: String
  active: Boolean
  marketingConsent: String
  identityCheck: String
  communications: [ContactCommunication!]
  source: ContactSource
  homePhone: String
  workPhone: String
  mobilePhone: String
  email: String
  primaryAddress: [ContactAddress!]
  secondaryAddress: [ContactAddress!]
  workAddress: [ContactAddress!]
  officeIds: [String!]
  negotiatorIds: [String!]
  metadata: JSON
  _eTag: String
  _links: JSON
  _embedded: JSON
}

type Contacts {
  _embedded: [Contact!]
  pageNumber: Int
  pageSize: Int
  pageCount: Int
  totalCount: Int
  _links: JSON
}

# Input Types
input ContactCommunicationInput {
  label: String
  detail: String
}

input ContactAddressInput {
  type: String
  buildingName: String
  buildingNumber: String
  line1: String
  line2: String
  line3: String
  line4: String
  postcode: String
  countryId: String
}
