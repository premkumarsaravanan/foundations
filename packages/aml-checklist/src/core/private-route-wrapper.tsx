import * as React from 'react'
import { Redirect, RouteComponentProps } from 'react-router-dom'
import { connect } from 'react-redux'
import { withRouter } from 'react-router'
import { ReduxState } from '@/types/core'
import { Loader, AppNavContainer } from '@reapit/elements'
import { RefreshParams, getTokenFromQueryString } from '@reapit/cognito-auth'
import Menu from '@/components/ui/menu'
import Routes from '@/constants/routes'
import { selectUserLoginStatus } from '@/selectors/auth'
import { Dispatch } from 'redux'
import { authSetRefreshSession } from '../actions/auth'

const { Suspense } = React

export interface PrivateRouteWrapperConnectActions {
  setRefreshSession: (refreshParams: RefreshParams) => void
}

export interface PrivateRouteWrapperConnectState {
  hasSession: boolean
  isDesktopMode: boolean
}

export type PrivateRouteWrapperProps = PrivateRouteWrapperConnectState &
  PrivateRouteWrapperConnectActions &
  RouteComponentProps & {
    path: string
  }

export const PrivateRouteWrapper: React.FunctionComponent<PrivateRouteWrapperProps> = ({
  children,
  hasSession,
  isDesktopMode,
  setRefreshSession
}) => {
  const desktopLogin = getTokenFromQueryString(location.search)

  if (desktopLogin && !isDesktopMode) {
    setRefreshSession(desktopLogin)
  }

  if (!hasSession) {
    return <Redirect to={Routes.LOGIN} />
  }

  return (
    <AppNavContainer>
      <Menu />
      <Suspense fallback={<Loader body />}>{children}</Suspense>
    </AppNavContainer>
  )
}

const mapStateToProps = (state: ReduxState): PrivateRouteWrapperConnectState => ({
  hasSession: selectUserLoginStatus(state),
  isDesktopMode: state?.auth?.refreshSession?.mode === 'DESKTOP'
})

const mapDispatchToProps = (dispatch: Dispatch): PrivateRouteWrapperConnectActions => ({
  setRefreshSession: refreshParams => dispatch(authSetRefreshSession(refreshParams))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(PrivateRouteWrapper))
