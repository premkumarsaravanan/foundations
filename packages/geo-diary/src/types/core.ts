import Routes from '@/constants/routes'
import ActionTypes from '@/constants/action-types'
import { OnlineState } from '@/reducers/online'
import { ErrorState } from '@/reducers/error'
import { HomeState } from '@/reducers/home'
import { AppointmentsState } from '@/reducers/appointments'
import { AppointmentDetailState } from '@/reducers/appointment-detail'
import { AuthState } from '@/reducers/auth'
import { NextAppointmentState } from '@/reducers/next-appointment'
import { DirectionState } from '@/reducers/direction'
import {
  PagedResultAppointmentModel_,
  AppointmentModel,
  OfficeModel,
  PropertyModel,
  NegotiatorModel
} from '@reapit/foundations-ts-definitions'

export type ExtendedPagedResultAppointmentModel_ = Omit<PagedResultAppointmentModel_, '_embedded'> & {
  _embedded?: ExtendedAppointmentModel[]
}

export type ExtendedAppointmentModel = AppointmentModel & {
  offices?: OfficeModel[]
  property?: PropertyModel
  negotiators?: NegotiatorModel[]
}

export interface Action<T> {
  readonly type: ActionType
  readonly data: T
}

export interface ActionCreator<T> {
  readonly type: string
  (data: T): Action<T>
}

export interface StringMap {
  [key: string]: string
}

export type PartialRecord<K extends keyof any, T> = { [P in K]?: T }

export type RouteValue = keyof typeof Routes

export type ActionType = keyof typeof ActionTypes

export type FormState = 'PENDING' | 'DONE' | 'SUBMITTING' | 'ERROR' | 'SUCCESS'

export interface FetcherParams<T> {
  method: 'GET' | 'POST' | 'PUT' | 'DELETE'
  api: string
  url: string
  headers: StringMap
  isPrivate?: boolean
  body?: T
}

export interface ReduxState {
  online: OnlineState
  error: ErrorState
  home: HomeState
  direction: DirectionState
  appointments: AppointmentsState
  appointmentDetail: AppointmentDetailState
  auth: AuthState
  nextAppointment: NextAppointmentState
}
