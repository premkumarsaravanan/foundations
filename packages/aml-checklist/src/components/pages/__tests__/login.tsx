import * as React from 'react'
import { shallow } from 'enzyme'
import {
  Login,
  LoginProps,
  renderForm,
  onSubmitHandler,
  LoginFormValues,
  mapStateToProps,
  mapDispatchToProps,
  resetSubmitting,
} from '@/components/pages/login'
import { ReduxState } from '@/types/core'
import { LoginParams } from '@reapit/cognito-auth'

const props = (hasSession: boolean): LoginProps => ({
  error: false,
  hasSession,
  login: jest.fn(),
  // @ts-ignore: just pick the needed props for the test
  history: {},
})

describe('Login', () => {
  it('should match a snapshot', () => {
    expect(shallow(<Login {...props(false)} />)).toMatchSnapshot()
  })

  it('should match a snapshot when hasSession', () => {
    expect(shallow(<Login {...props(true)} />)).toMatchSnapshot()
  })

  describe('renderForm', () => {
    it('should match a snapshot when ERROR true', () => {
      const fn = renderForm({ isSubmitting: false, error: true })()
      expect(fn).toMatchSnapshot()
    })

    it('should match a snapshot when ERROR true', () => {
      const fn = renderForm({ isSubmitting: false, error: false })()
      expect(fn).toMatchSnapshot()
    })
  })

  describe('onSubmitHandler', () => {
    const mockProps = {
      setIsSubmitting: jest.fn(),
      login: jest.fn(),
    }
    const mockValues = {} as LoginFormValues
    it('should run correctly', () => {
      onSubmitHandler(mockProps)(mockValues)
      expect(mockProps.setIsSubmitting).toBeCalledWith(true)
      expect(mockProps.login).toBeCalledWith({ ...mockValues, loginType: 'CLIENT' })
    })
  })

  describe('resetSubmitting', () => {
    const setIsSubmitting = jest.fn()
    it('should run correctly', () => {
      resetSubmitting(true, setIsSubmitting)()
      expect(setIsSubmitting).toBeCalledWith(false)
    })
  })

  describe('mapStateToProps', () => {
    it('should run correctly', () => {
      // @ts-ignore: only pick necessary props
      const mockState = {
        auth: {
          refreshSession: null,
          loginSession: null,
          error: false,
        },
      } as ReduxState
      const result = mapStateToProps(mockState)
      expect(result).toEqual({
        hasSession: false,
        error: false,
      })
    })
  })

  describe('mapDispatchToProps', () => {
    it('should render correctly', () => {
      const mockDispatch = jest.fn()
      const { login } = mapDispatchToProps(mockDispatch)
      login({} as LoginParams)
      expect(mockDispatch).toBeCalled()
    })
  })
})
