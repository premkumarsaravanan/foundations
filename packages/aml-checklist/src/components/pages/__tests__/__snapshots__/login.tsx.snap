// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renderForm should match a snapshot when ERROR true 1`] = `
<ForwardRef
  data-test="login-form"
>
  <Input
    dataTest="login-email"
    id="email"
    labelText="Email"
    name="email"
    placeholder="name@address.com"
    type="email"
  />
  <Input
    dataTest="login-password"
    id="password"
    labelText="Password"
    name="password"
    placeholder="Enter your password"
    type="password"
  />
  <Level>
    <Button
      disabled={false}
      loading={false}
      type="submit"
      variant="primary"
    >
      Login
    </Button>
  </Level>
  <Alert
    message="Login failed, user credentials not recognised"
    type="danger"
  />
</ForwardRef>
`;

exports[`Login renderForm should match a snapshot when ERROR true 2`] = `
<ForwardRef
  data-test="login-form"
>
  <Input
    dataTest="login-email"
    id="email"
    labelText="Email"
    name="email"
    placeholder="name@address.com"
    type="email"
  />
  <Input
    dataTest="login-password"
    id="password"
    labelText="Password"
    name="password"
    placeholder="Enter your password"
    type="password"
  />
  <Level>
    <Button
      disabled={false}
      loading={false}
      type="submit"
      variant="primary"
    >
      Login
    </Button>
  </Level>
</ForwardRef>
`;

exports[`Login should match a snapshot 1`] = `
<div>
  <div
    className="undefined false"
  >
    <H1
      isCentered={true}
    >
      Sign in
    </H1>
    <p
      className="pb-8"
    >
      Welcome to Reapit Online Check List
    </p>
    <Formik
      initialValues={
        Object {
          "email": "",
          "password": "",
        }
      }
      onSubmit={[Function]}
      validate={[Function]}
    >
      <Component />
    </Formik>
  </div>
  <div>
    <img
      alt="Reapit Graphic"
      src=""
    />
  </div>
</div>
`;

exports[`Login should match a snapshot when hasSession 1`] = `
<Redirect
  to="/"
/>
`;
